mypy/build.py

```diff
@@ -897,23 +960,35 @@ def find() -> Optional[str]:
             # Prefer package over module, i.e. baz/__init__.py* over baz.py*.
             for extension in PYTHON_EXTENSIONS:
                 path = base_path + sepinit + extension
+                path_stubs = base_path + '-stubs' + sepinit + extension
                 if is_file(path) and verify_module(id, path):
                     return path
+                elif is_file(path_stubs) and verify_module(id, path_stubs):
+                    return path_stubs
             # No package, look for module.
             for extension in PYTHON_EXTENSIONS:
                 path = base_path + extension
                 if is_file(path) and verify_module(id, path):
                     return path
         return None
 
-    key = (id, lib_path)
```

Comment:
I removed it because the lib_path never changes, so it was worthless to use it as a key. Also FWIW, Guido is going to be refactoring a lot of the machinery here (see https://github.com/python/mypy/issues/4365) after this and namespace support are merged.

Reply:
The problem is, `lib_path` never changes until someone decides that it does, and then your cache is broken. What if someone calls `mypy.main` twice with different `lib_path` or `python_executable` arguments? Now your cache is garbage.

Caching is hard, `@functools.lru_cache` is easy. The cache here is to save filesystem costs, so you can easily afford a few extra `hash()` calls